src/ctim/domain/id.cljc

```diff
@@ -131,12 +131,15 @@
     (nth (re-matches long-id-re s) 8)
     s))
 
-(defn short-id->long-id [short-id url-params-fn]
-  (when-let [type (nth (re-matches short-id-re short-id) 2)]
-    (long-id
-     (short-id->id type
-                   short-id
-                   (url-params-fn)))))
+(s/defn short-id->long-id :- (s/maybe s/Str)
+  [short-id :- (s/maybe s/Str)
+   url-params-fn :- (s/maybe (s/=> {s/Keyword s/Any}))]
+  (when short-id
+    (when-let [type (nth (re-matches short-id-re short-id) 2)]
```

Comment:
you could remove first `when` and use [`not-found`parameter of `nth`](https://clojuredocs.org/clojure.core/nth):
`nth throws an exception unless not-found is supplied.`
```clojure
(when-let [type (nth (re-matches short-id-re short-id) 2 nil)]
...
)
```